pkg/cloud/converters/tags.go

```diff
@@ -64,6 +66,9 @@ func MapToTags(src infrav1.Tags) []*ec2.Tag {
 		tags = append(tags, tag)
 	}
 
+	// Sort so that unit tests can expect a stable order
+	sort.Slice(tags, func(i, j int) bool { return *tags[i].Key < *tags[j].Key })
```

Comment:
Could we make `Tags` an alias sliced and satisfy the sort interface instead of copying this function over?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/1059821a5382f5c57de29b8b8e7cc4ba344f8d3b/api/v1beta2/tags.go#L30

Reply:
I like the `SortByKey` function idea; it saves on a bigger refactoring.

However, I do think that revisiting these functions and types for the Tags makes sense.

Reply:
`slices.SortFunc` won't work because Go generics do not yet support accessing common fields.

Can we leave this for another issue since it blows up the scope of this PR and the problem is also in existing code?